### ISSUE TITLE
Using --omit with -b can confuse the resets the throttling calculation 

### ISSUE THREAD

**Issue by joemce** [esnet/iperf#419]
If --omit is combined with -b, the byte counters used for the throttling check are reset. This will throw off the throttling check when the omit timer fires. See the example below. The bandwidth in the final interval before the omit timer jumps from 1 to 2 M, and the first interval after the omit displays a bandwidth of 0.

``` text
$ ./iperf3 -v
iperf 3.1
Linux galway 4.4.0-8-generic #23-Ubuntu SMP Wed Feb 24 20:45:30 UTC 2016 x86_64
Optional features available: CPU affinity setting, IPv6 flow label, TCP congestion algorithm setting, sendfile / zerocopy
```

``` text
$ ./iperf3 -c 127.0.0.1 -p 3000 -b 1M --omit 8 -t 1 --reverse
Connecting to host 127.0.0.1, port 3000
Reverse mode, remote host 127.0.0.1 is sending
[  4] local 127.0.0.1 port 53438 connected to 127.0.0.1 port 3000
[ ID] Interval           Transfer     Bandwidth
[  4]   0.00-1.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   1.00-2.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   2.00-3.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   3.00-4.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   4.00-5.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   5.00-6.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   6.00-7.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   7.00-8.00   sec   256 KBytes  2.10 Mbits/sec                  (omitted)
[  4]   0.00-1.00   sec  0.00 Bytes  0.00 bits/sec                  
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[  4]   0.00-1.00   sec   128 KBytes  1.05 Mbits/sec    0             sender
[  4]   0.00-1.00   sec  0.00 Bytes  0.00 bits/sec                  receiver

iperf Done.
```

**Comment by joemce** [comment]
I updated my previous pull request to include a possible fix for --omit with -b.

https://github.com/esnet/iperf/pull/414/commits/1f240f31f7d8a8c69f0bcbc4d8e058d6e459af0c

By tracking the omit bytes and not reseting the sent_bytes, the throttle calculation can continue to work across the omit timer.

``` text
./src/iperf3 -c 127.0.0.1 -p 3000 -b 1M --omit 8 -t 1 --reverse
Connecting to host 127.0.0.1, port 3000
Reverse mode, remote host 127.0.0.1 is sending
[  4] local 127.0.0.1 port 53488 connected to 127.0.0.1 port 3000
[ ID] Interval           Transfer     Bandwidth
[  4]   0.00-1.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   1.00-2.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   2.00-3.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   3.00-4.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   4.00-5.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   5.00-6.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   6.00-7.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   7.00-8.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   0.00-1.00   sec   128 KBytes  1.05 Mbits/sec                  
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[  4]   0.00-1.00   sec   128 KBytes  1.05 Mbits/sec    0             sender
[  4]   0.00-1.00   sec   128 KBytes  1.05 Mbits/sec                  receiver

iperf Done.
```

**Comment by bmah888** [comment]
Assign to @bltierney for investigation.

**Comment by bmah888** [comment]
@joemce:  Is this bug fixed by your pull request #414, which I merged to esnet:master last year?

**Comment by bltierney** [comment]
I'm not sure if this patch is now in master or not, but I've confirmed that this is still a problem in both normal and reverse mode.

For example:

./iperf3 -c nettest.lbl.gov -p 5300 -fq-rate 1M --omit 4 -t 1 --reverse
 
[  4]   0.00-1.00   sec  64.6 MBytes   542 Mbits/sec                  (omitted)
[  4]   1.00-2.00   sec  67.1 MBytes   563 Mbits/sec                  (omitted)
[  4]   2.00-3.00   sec  67.1 MBytes   563 Mbits/sec                  (omitted)
[  4]   3.00-4.00   sec  87.3 MBytes   732 Mbits/sec                  (omitted)
[  4]   0.00-1.00   sec  69.3 MBytes   581 Mbits/sec
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[  4]   0.00-1.00   sec  73.1 MBytes   613 Mbits/sec    1             sender
[  4]   0.00-1.00   sec  69.8 MBytes   586 Mbits/sec                  receiver

and:

./iperf3 -c nettest.lbl.gov -p 5300 -fq-rate 1M --omit 4 -t 1

[  4]   0.00-1.00   sec  11.7 MBytes  98.0 Mbits/sec    0    173 KBytes       (omitted)
[  4]   1.00-2.00   sec  33.8 MBytes   284 Mbits/sec    0    351 KBytes       (omitted)
[  4]   2.00-3.00   sec  58.3 MBytes   489 Mbits/sec    0    577 KBytes       (omitted)
[  4]   3.00-4.00   sec   101 MBytes   849 Mbits/sec    0   1003 KBytes       (omitted)
[  4]   0.00-1.00   sec   115 MBytes   963 Mbits/sec    0   1.42 MBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[  4]   0.00-1.00   sec   115 MBytes   964 Mbits/sec    0             sender
[  4]   0.00-1.00   sec   117 MBytes   983 Mbits/sec                  receiver

**Comment by bmah888** [comment]
The aformentioned pull request #414 is on master and the 3.1-STABLE branch.

I see a couple of problems now I've stared at the last set of test output:

1.  FQ wasn't configured correctly in those command lines.  It's `--fq-rate`, not `-fq-rate` (note number of dashes.  Clearly iperf3's command-line parser needs to handle that better (like "throw an error"), because `q` isn't even valid following `-f`.

2.  The `--fq-rate` option (as opposed to `--bandwidth`) doesn't actually exercise this problem because `--fq-rate` doesn't use any of iperf3's internal pacing mechanisms, and the original bug had to do with that.

3.  I'm not sure that `-t 1` is appropriate...isn't that kind of a short measurement interval?

Here's my version of the above two tests:
```
[bmah@antg-staging iperf]$ src/iperf3 --client antg-dev --port 5202 --bandwidth 1m --omit 4 --time 5
Connecting to host antg-dev, port 5202
[  4] local 2001:400:210:151::25 port 57264 connected to 2001:400:210:151::11 port 5202
[ ID] Interval           Transfer     Bandwidth       Retr  Cwnd
[  4]   0.00-1.00   sec   176 KBytes  1.44 Mbits/sec    0   16.7 KBytes       (omitted)
[  4]   1.00-2.00   sec  72.5 KBytes   594 Kbits/sec    0   18.1 KBytes       (omitted)
[  4]   2.00-3.00   sec   155 KBytes  1.27 Mbits/sec    0   16.7 KBytes       (omitted)
[  4]   3.00-4.00   sec   152 KBytes  1.25 Mbits/sec    0   26.5 KBytes       (omitted)
[  4]   0.00-1.00   sec  89.2 KBytes   731 Kbits/sec    0   25.1 KBytes
[  4]   1.00-2.00   sec   178 KBytes  1.46 Mbits/sec    0   26.5 KBytes
[  4]   2.00-3.00   sec  93.4 KBytes   765 Kbits/sec    0   30.7 KBytes
[  4]   3.00-4.00   sec  93.4 KBytes   765 Kbits/sec    0   32.1 KBytes
[  4]   4.00-5.00   sec   128 KBytes  1.05 Mbits/sec    0   36.3 KBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[  4]   0.00-5.00   sec   583 KBytes   955 Kbits/sec    0             sender
[  4]   0.00-5.04   sec   583 KBytes   948 Kbits/sec                  receiver

iperf Done.
```
```
[bmah@antg-staging iperf]$ src/iperf3 --client antg-dev --port 5202 --bandwidth 1m --omit 4 --time 5 --reverse
Connecting to host antg-dev, port 5202
Reverse mode, remote host antg-dev is sending
[  4] local 2001:400:210:151::25 port 57544 connected to 2001:400:210:151::11 port 5202
[ ID] Interval           Transfer     Bandwidth
[  4]   0.00-1.00   sec   153 KBytes  1.26 Mbits/sec                  (omitted)
[  4]   1.00-2.00   sec   112 KBytes   914 Kbits/sec                  (omitted)
[  4]   2.00-3.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   3.00-4.00   sec   128 KBytes  1.05 Mbits/sec                  (omitted)
[  4]   0.00-1.00   sec   126 KBytes  1.03 Mbits/sec
[  4]   1.00-2.00   sec   128 KBytes  1.05 Mbits/sec
[  4]   2.00-3.00   sec   128 KBytes  1.05 Mbits/sec
[  4]   3.00-4.00   sec   126 KBytes  1.03 Mbits/sec
[  4]   4.00-5.00   sec   128 KBytes  1.05 Mbits/sec
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[  4]   0.00-5.04   sec   635 KBytes  1.03 Mbits/sec    0             sender
[  4]   0.00-5.00   sec   635 KBytes  1.04 Mbits/sec                  receiver

iperf Done.
```
This looks somewhat more sane to me, and in fact is roughly consistent with testing output from @joemce about a year ago.

I'm going to claim this was fixed by pull request #414.  Closing this issue, but it can be reopened if needed.
